confgenerator/resourcedetector/detector.go

```diff
@@ -22,17 +22,37 @@ import (
 // An implementation of the Resource interface will have fields represent
 // available attributes about the current monitoring resource.
 type Resource interface {
+	ProjectName() string
+	IsAutoDetected() bool
```

Comment:
this shouldn't be part of the Resource interface; maybe it needs to be tracked somewhere (in `Platform`, perhaps?) but each resource shouldn't have to know about it.